### ISSUE TITLE
Native Teams Templates support

### ISSUE THREAD

**Issue by alexc-MSFT** [OfficeDev/microsoft-teams-apps-requestateam#179]
**Summary**
With this release the app template will integrate with the native Teams Templates functionality that is now available (templates defined from the Teams Admin Centre). 

![image](https://user-images.githubusercontent.com/12395485/96421683-f548a700-11ee-11eb-87bd-ff1cde655f2a.png)

**Intended Outcome**
To be able to integrate with the new native templates functionality.

**How will it work**
You will be able to define templates in the Teams Admin Centre, these will be synced using the Microsoft Graph to the 'Teams Templates' list. The list will be kept in constant sync in case templates are deleted.

The Power App will read from the above list and allow users to select a template.

The provisioning will be updated to ensure it can provision from defined templates in addition to Microsoft provided templates e.g. Adopt O365.

**Please note - This functionality has a dependency on the introduction of new Graph endpoints to surface templates defined in the Admin Centre.**

**Comment by iehlers** [comment]
How we can update to latest version on an existing integrated teams-apps-requestateam?

**Comment by jcmmarq** [comment]
yes really needed functionality as we are using our own templates today

**Comment by jcmmarq** [comment]
how do we upgrade??
